apps/wallet-mobile/src/navigation.tsx

```diff
@@ -514,6 +536,24 @@ export const useWalletNavigation = () => {
   } as const).current
 }
 
+export const useNavigateToWalletSelectionSync = () => {
+  const {resetToWalletSetup, resetToWalletSelection} = useWalletNavigation()
+  const walletManager = useWalletManager()
+  const {refetch} = useHasWallets(walletManager, {
+    enabled: false,
+    onSuccess: (hasWallets) => {
+      if (hasWallets) {
+        resetToWalletSelection()
+        return
+      }
+
+      resetToWalletSetup()
+    },
+  })
+
+  return refetch
+}
+
```

Comment:
To determine if the user has wallets, it is necessary to use the listWallets method.

When deleting a wallet, its metadata is not deleted at that time, but rather when the app is launched. When using the listWallets method, a check is performed to filter the IDs of the wallets that are going to be deleted.

In short, it is necessary to perform a refetch to obtain an updated value of hasWallets, that is needed to determine if we are going to show the wallets list screen or the setup wallet screen when a wallet is removed.

Reply:
Roger, but this will be way simpler to achieve with the changes in the walletManager, (we can't pick right now), create a ticket for 4.28 to refactor it

Reply:
Question invalidating the query cache didn't do the job? It should, can listWallets mind for deleted ones.

Reply:
i fixed the queryKey [here](https://github.com/Emurgo/yoroi/pull/3215/commits/6f0aec71c20595e301fc9f52be693581ddd4e263#diff-74c3148fcce592101f4de41e6d2c636df4d4c9145956b49662585dbece349b40L703) but it doesn't work yet : (

Reply:
FIXED.

The problem was in using incorrect queryId to invalidate.